user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Cart Events
+taxonomy:
+    category: docs
+---
+
+Drupal Commerce's cart system provides a full range of events for your custom functionality and modules to react to.
+
+The events use Drupal's standard [Symphony based event system](https://www.drupal.org/docs/8/creating-custom-modules/subscribe-to-and-dispatch-events#s-event-systems-overview). So a solid understanding of that is useful. Events are defined in `\Drupal\commerce_cart\Event\CartEvents`.
+
+Events provided:
+- `CartEvents::CART_EMPTY`: Fired after emptying the cart.
+- `CartEvents::CART_ENTITY_ADD`: Fired after adding a purchaseable entity to the cart.
+- `CartEvents::CART_ORDER_ITEM_UPDATE`: Fired after an order item.
+- `CartEvents::CART_ORDER_ITEM_REMOVE`: Fired after removing an order item form the cart.
+- `CartEvents::CART_ITEM_COMPARISON_FIELDS`: Fired when altering the list of comparison fields.
+
+## Basic Example
+
+This adds a related product when a specified product is added. A reference field - ala commerce_pado is probably better under most situations.
```

Comment:
Explanation here is just a little confusing. Maybe change to, "This adds a related product when a specific product is added. In this example, we just hardcode the entity ID of a specific variation. In most situations, you'd probably want to add an entity reference field to your product variations to allow admin users to dynamically specify the "related" product for each variation. For an example, see the [Commerce Product Add-on](https://www.drupal.org/project/commerce_pado) contrib module."